fpdf/fpdf.py

```diff
@@ -2990,11 +2979,16 @@ def _render_styled_text_line(
                     frag.font_size,
                     link,
                 )
+
         if sl:
             # If any PDF settings have been left modified, wrap the line in a local context.
             if (
-                current_font_style != self.font_style
+                current_ws != 0.0
+                or current_font != self.current_font
+                or current_text_mode != self.text_mode
                 or self.fill_color != self.text_color
+                or current_font_stretching != self.font_stretching
+                # or current_char_spacing != self.char_spacing
```

Comment:
Why this comment-out line?

Reply:
I thought it might make some files smaller, but now I actually think it's a mistake. Will fix.